### PR TITLE
KIALI-727 Migrate from npm to yarn

### PR DESCRIPTION
**Pull request by josejulio**
We were having some issues with `npm` and `package-lock.json` and we aren't alone here, there is an [issue]( https://github.com/npm/npm/issues/17722#issue-242057316) in npm's github and some projects have moved to yarn.

See https://github.com/patternfly/patternfly-react/pull/289#issuecomment-376558383 for good reasons of moving from npm to yarn. 

After merging this we would need to use yarn instead of npm ~~, still need to update the Readme.~~

Todo:
- [x] Update Readme
- [x] Test with OSX

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,8 +17,8 @@
     "deep-freeze": "0.0.1",
     "js-yaml": "3.11.0",
     "lodash": "4.17.10",
-    "patternfly": "3.45.3",
-    "patternfly-react": "2.3.0",
+    "patternfly": "3.46.1",
```
Comment (mtho11) [inline package.json]: Updating a lot of npm packages that are not related to this `yarn` PR. Or are they?

**Comment by jmazzitelli** [comment]
I just want to confirm that this is still OK (kiali server-side pulling the kiali UI package from npm repo):

https://github.com/kiali/kiali/blob/master/Makefile#L147-L154

**Comment by mtho11** [comment]
Yes, @jmazzitelli you have my word on it ;)

File: .travis.yml [hunk 2]
```diff
@@ -8,18 +8,21 @@ addons:
 node_js:
   - 8
 cache:
+  yarn: true
   directories:
     - node_modules
 script:
-  - npm run prettier -- --list-different &&
-    npm run build-css &&
```
Comment (josejulio) [inline .travis.yml]: I'm removing this, because build already includes this command

**Comment by josejulio** [comment]
@mtho11 this is ready to be reviewed.
Rebased, squashed commits, tested on mac and fedora.

File: .travis.yml [hunk 3]
```diff
@@ -8,18 +8,22 @@ addons:
 node_js:
   - 8
 cache:
+  yarn: true
   directories:
     - node_modules
+before_install:
+  yarn --frozen-lockfile --non-interactive || true
```
Comment (mtho11) [inline .travis.yml]: I didn't know about this yarn option.
